sklearnex/tests/_utils.py

```diff
@@ -135,8 +135,10 @@ def gen_dataset(estimator, queue=None, target_df=None, dtype=np.float64):
     else:
         raise ValueError("Unknown dataset type")
 
-    X = _convert_to_dataframe(X, sycl_queue=queue, target_df=target_df, dtype=dtype)
-    y = _convert_to_dataframe(y, sycl_queue=queue, target_df=target_df, dtype=dtype)
+    if queue is not None:
+        X = _convert_to_dataframe(X, sycl_queue=queue, target_df=target_df, dtype=dtype)
+        y = _convert_to_dataframe(y, sycl_queue=queue, target_df=target_df, dtype=dtype)
+
```

Comment:
I'm pretty sure that the test was red without this edit. I'll recheck and create the task for investigation if it is still true

Reply:
![image](https://github.com/intel/scikit-learn-intelex/assets/102592747/dc5539ef-cfe2-4806-af0d-7c4cd2381460)
the problem still exists. _convert_to_dataframe changes data here if called with queue=None and it lead to failure. I will add TODO comment and try to do shorter reproducer for the future task. I guess it would be better to leave the logic here as is for now.